feat(plugins): add datadog tracing plugin

mayocream opened the pull request:
### Description

Add docs for the new Datadog Tracing plugin.

KAG-454

### Checklist 

- [x] Review label added <!-- (see below) -->
- [x] PR pointed to correct branch (`main` for immediate publishing, or a release branch: e.g. `release/gateway-3.2`, `release/deck-1.17`)


<!-- !!! Only Kong employees can add labels due to a GitHub limitation. If you're an OSS contributor, thank you! The maintainers will label this PR for you !!! -->

<!-- When raising a pull request, indicate what type of review you need with one of the following labels:

    review:copyedit: Request for writer review.
    review:general: Review for general accuracy and presentation. Does the doc work? Does it output correctly?
    review:tech: Request for technical review for a docs platform change.
    review:sme: Request for review from an SME (engineer, PM, etc).

At least one of these labels must be applied to a PR or the build will fail.
-->



[DOCU-2918]: https://konghq.atlassian.net/browse/DOCU-2918?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

colinh-kong commented:
Needs a header image
![image](https://user-images.githubusercontent.com/120141901/212914568-8fe8421d-f948-4081-b3c8-5c2977902e5d.png)

Guaris commented:
@mayocream 

Hey mayo, after the parameters this doc is a copy of the open telemetry doc. Could you write a usage section specific to the datadog tracing plugin?

mayocream commented:
> @mayocream
> 
> Hey mayo, after the parameters this doc is a copy of the open telemetry doc. Could you write a usage section specific to the datadog tracing plugin?

It was a draft, I updated the PR and rebased it, it's ready for review.

fffonion commented on app/_hub/kong-inc/datadog-tracing/_index.md:
````diff
@@ -0,0 +1,177 @@
+
+```bash
+curl -X POST http://<admin-hostname>:8001/plugins \
+    --data "name=Datadog-tracing"  \
````
```suggestion
    --data "name=datadog-tracing"  \
```

lena-larionova commented on app/_hub/kong-inc/datadog-tracing/_index.md:
```diff
@@ -0,0 +1,175 @@
+
+Enable the OpenTelemetry tracing capability in {{site.base_gateway}}'s configuration:
+
+- `tracing_instrumentations = request`: valid values can be found in the [tracing instrumentations](/gateway/latest/reference/configuration/#tracing_instrumentations) section of the configuration documentation.
```
```suggestion
- `tracing_instrumentations = request`: Valid values can be found in the [tracing instrumentations](/gateway/latest/reference/configuration/#tracing_instrumentations) section of the configuration documentation.
```

lena-larionova commented on app/_hub/kong-inc/datadog-tracing/_index.md:
```diff
@@ -0,0 +1,175 @@
+Enable the OpenTelemetry tracing capability in {{site.base_gateway}}'s configuration:
+
+- `tracing_instrumentations = request`: valid values can be found in the [tracing instrumentations](/gateway/latest/reference/configuration/#tracing_instrumentations) section of the configuration documentation.
+- `tracing_sampling_rate = 1.0`: tracing instrumentation sampling rate.
```
```suggestion
- `tracing_sampling_rate = 1.0`: Tracing instrumentation sampling rate.
```

lena-larionova commented on app/_hub/kong-inc/datadog-tracing/_index.md:
````diff
@@ -0,0 +1,175 @@
+### Set up the Datadog Agent
+
+```bash
+docker run -d -p 8126:8126 \
+              --cgroupns host \
+              --pid host \
+              -v /var/run/docker.sock:/var/run/docker.sock:ro \
+              -v /proc/:/host/proc/:ro \
+              -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro \
+              -e DD_API_KEY=<api-key> \
+              -e DD_APM_NON_LOCAL_TRAFFIC=true \
+              -e DD_APM_ENABLED=true \
+              gcr.io/datadoghq/agent:latest
````
```suggestion
docker run -d -p 8126:8126 \
  --cgroupns host \
  --pid host \
  -v /var/run/docker.sock:/var/run/docker.sock:ro \
  -v /proc/:/host/proc/:ro \
  -v /sys/fs/cgroup/:/host/sys/fs/cgroup:ro \
  -e DD_API_KEY=<example-api-key> \
  -e DD_APM_NON_LOCAL_TRAFFIC=true \
   -e DD_APM_ENABLED=true \
   gcr.io/datadoghq/agent:latest
```
No need for such deep indents.

lena-larionova commented on app/_hub/kong-inc/datadog-tracing/_index.md:
```diff
@@ -0,0 +1,175 @@
+  This allows you to set the sampling rate to a lower value to reduce the impact of the instrumentation on {{site.base_gateway}}'s proxy performance in production.
+
+### Set up the Datadog Agent
+
```
Needs an intro to the section. Something like:

```suggestion

Set up the Datadog Agent. You can use the following example Docker configuration, substituting the `example-api-key` with your own key:

```

lena-larionova commented on app/_hub/kong-inc/datadog-tracing/_index.md:
````diff
@@ -0,0 +1,175 @@
+              gcr.io/datadoghq/agent:latest
+```
+
+For more information about installing the Datadog Agent on Docker, see [Datadog's documentation](https://docs.datadoghq.com/agent/docker/apm/?tab=linux). If you want to install another container option you can find the instructions [on this list](https://docs.datadoghq.com/containers/).
````
```suggestion
For more information about installing the Datadog Agent on Docker, see [Datadog's documentation](https://docs.datadoghq.com/agent/docker/apm/?tab=linux). If you want to install another container option, you can find the instructions in [Datadog's container list](https://docs.datadoghq.com/containers/).
```
We should use less generic link titles for better SEO and accessibility.

lena-larionova commented on app/_hub/kong-inc/datadog-tracing/_index.md:
````diff
@@ -0,0 +1,175 @@
+    --data "config.environment=prod"
+```
+
+## OpenTelemetry Functionality
````
```suggestion
## OpenTelemetry functionality
```

lena-larionova commented on app/_hub/kong-inc/datadog-tracing/_index.md:
```diff
@@ -0,0 +1,175 @@
+
+### Propagation
+
+The OpenTelemetry plugin propagates the following headers:
```
```suggestion
The Datadog Tracing plugin propagates the following headers:
```

lena-larionova commented on app/_hub/kong-inc/datadog-tracing/_index.md:
```diff
@@ -0,0 +1,175 @@
+
+The Datadog Tracing spans are printed to the console when the log level is set to `debug` in the Kong configuration file.
+
+An example of debug logs output:
```
```suggestion
Here's an example of the debug log output:
```

lena-larionova commented on app/_hub/kong-inc/datadog-tracing/_index.md:
```diff
@@ -0,0 +1,175 @@
+
+### Known issues
+
+- Datadog tracing Only supports the HTTP protocols (http/https) of {{site.base_gateway}}.
```
```suggestion
- Datadog Tracing only supports the HTTP protocols (`http`/`https`) of {{site.base_gateway}}.
```

lena-larionova commented on app/_hub/kong-inc/datadog-tracing/_index.md:
```diff
@@ -0,0 +1,175 @@
+### Known issues
+
+- Datadog tracing Only supports the HTTP protocols (http/https) of {{site.base_gateway}}.
+- May impact the performance of {{site.base_gateway}}.
```
```suggestion
- This plugin may impact the performance of {{site.base_gateway}}.
```

lena-larionova commented on app/_hub/kong-inc/datadog-tracing/_index.md:
```diff
@@ -0,0 +1,175 @@
+
+- Datadog tracing Only supports the HTTP protocols (http/https) of {{site.base_gateway}}.
+- May impact the performance of {{site.base_gateway}}.
+  It's recommended to set the sampling rate (`tracing_sampling_rate`)
```
```suggestion
  We recommend setting the sampling rate (`tracing_sampling_rate`)
```

lena-larionova commented on app/_hub/kong-inc/datadog-tracing/_index.md:
```diff
@@ -0,0 +1,175 @@
+- Datadog tracing Only supports the HTTP protocols (http/https) of {{site.base_gateway}}.
+- May impact the performance of {{site.base_gateway}}.
+  It's recommended to set the sampling rate (`tracing_sampling_rate`)
+  via Kong configuration file when using the Datadog Tracing plugin.
```
```suggestion
  in the {{site.base_gateway}} configuration file when using the Datadog Tracing plugin.
```

lena-larionova commented on app/_hub/kong-inc/datadog-tracing/versions.yml:
```diff
@@ -0,0 +1,7 @@
+releases:
+  minimum_version: '3.2.x'
+
+overrides:
+  3.2.x: 0.1.0
```
Why do we need an override for this plugin version? As far as I know, the plugin version _should_ match the Kong version starting in 3.0. If it doesn't, then something's wrong.

Guaris commented on app/_data/tables/plugin_index.yml:
```diff
@@ -439,6 +439,14 @@
+    
+    - name: Datadog Tracing
+      free: No
+      plus: Yes
```
```suggestion
      plus: No
```
